services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/model/CloudFrontSignerRequest.java

```diff
@@ -0,0 +1,217 @@
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.annotations.ThreadSafe;
+import software.amazon.awssdk.utils.builder.CopyableBuilder;
+import software.amazon.awssdk.utils.builder.ToCopyableBuilder;
+
+/**
+ * Request to generate a CloudFront signed URL or signed cookie
+ */
+@Immutable
+@ThreadSafe
+@SdkPublicApi
+public final class CloudFrontSignerRequest
+    implements ToCopyableBuilder<CloudFrontSignerRequest.Builder, CloudFrontSignerRequest> {
+
+    private final String resourceUrl;
+    private final PrivateKey privateKey;
+    private final String keyPairId;
+    private final Instant expirationDate;
+    private final Instant activeDate;
+    private final String ipRange;
+
+    private CloudFrontSignerRequest(DefaultBuilder builder) {
+        this.resourceUrl = builder.resourceUrl;
+        this.privateKey = builder.privateKey;
+        this.keyPairId = builder.keyPairId;
+        this.expirationDate = builder.expirationDate;
+        this.activeDate = builder.activeDate;
+        this.ipRange = builder.ipRange;
+    }
+
+    /**
+     * Create a builder that can be used to create a {@link CloudFrontSignerRequest}
+     */
+    public static Builder builder() {
+        return new DefaultBuilder();
+    }
+
+    @Override
+    public Builder toBuilder() {
+        return new DefaultBuilder(this);
+    }
+
+    /**
+     * Returns the resource URL
+     */
+    public String resourceUrl() {
+        return resourceUrl;
+    }
+
+    /**
+     * Returns the private key
+     */
+    public PrivateKey privateKey() {
+        return privateKey;
+    }
+
+    /**
+     * Returns the key pair ID
+     */
+    public String keyPairId() {
+        return keyPairId;
+    }
+
+    /**
+     * Returns the expiration date
+     */
+    public Instant expirationDate() {
+        return expirationDate;
+    }
+
+    /**
+     * Returns the active date
+     */
+    public Instant activeDate() {
+        return activeDate;
+    }
+
+    /**
+     * Returns the IP range
+     */
+    public String ipRange() {
+        return ipRange;
+    }
+
+    @NotThreadSafe
+    @SdkPublicApi
+    public interface Builder extends CopyableBuilder<CloudFrontSignerRequest.Builder, CloudFrontSignerRequest> {
+
+        /**
+         * Configure the resource URL to be signed
+         * <p>
+         *     The URL or path that uniquely identifies a resource within a
+         *     distribution. For standard distributions the resource URL will
+         *     be <tt>"http://" + distributionName + "/" + objectKey</tt>
+         *     (may also include URL parameters. For distributions with the
+         *     HTTPS required protocol, the resource URL must start with
+         *     <tt>"https://"</tt>
+         * </p>
+         */
+        Builder resourceUrl(String resourceUrl);
+
+        /**
+         * Configure the private key to be used to sign the policy
+         */
+        Builder privateKey(PrivateKey privateKey);
```

Comment:
Since we're not letting the users supply the file directly anymore, the javadoc should probably point to the load.. method in the utility class

Reply:
added @link